data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -236,6 +236,9 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
             final Map<String, Object> outputMap = new HashMap<>();
             final Event recordEvent = record.getData();
             final String groupsRaw = recordEvent.get(keyValueProcessorConfig.getSource(), String.class);
+            if (groupsRaw == null) {
```

Comment:
Thanks for the fix! 

Can we add `tagsOnFailure` to metadata in this case? Similar to Line 250 below.

Reply:
We could add it but I think that should be a separate PR.

Reply:
Will resolve this duplicate issue (https://github.com/opensearch-project/data-prepper/issues/3496). I think tagging it makes sense